pkg/controller/azvolume.go

```diff
@@ -576,9 +576,14 @@ func (c *SharedState) createAzVolumeFromPv(ctx context.Context, pv v1.Persistent
 		} else {
 			storageClass, err := c.kubeClient.StorageV1().StorageClasses().Get(ctx, pv.Spec.StorageClassName, metav1.GetOptions{})
 			if err != nil {
-				return fmt.Errorf("failed to get storage class (%s): %v", pv.Spec.StorageClassName, err)
+				if !errors.IsNotFound(err) {
+					return fmt.Errorf("failed to get storage class (%s): %v", pv.Spec.StorageClassName, err)
+				}
+				klog.Warningf("storage class (%s) for PV (%s) cannot be found, trying to get maxMountReplicaCount from VolumeAttributes.", pv.Spec.StorageClassName, pv.Name)
+				_, maxMountReplicaCount = azureutils.GetMaxSharesAndMaxMountReplicaCount(pv.Spec.CSI.VolumeAttributes, azureutils.IsMultiNodePersistentVolume(pv))
```

Comment:
Shouldn't these values always jut come from the `PV`? Is there a case where the `StorageClass` would have these parameters but not the `PV`?

Reply:
I originally thought "maybe migration scenario?" but on second thought, I don't think so. I think it is safe to remove fetching storageclass part at its entirety.